aries_cloudagent/resolver/base.py

```diff
@@ -59,11 +59,11 @@ def native(self):
         return self.type == ResolverType.NATIVE
 
     @property
-    @abstractmethod
     def supported_methods(self) -> Sequence[str]:
         """Return list of DID methods supported by this resolver."""
+        raise NotImplementedError()
 
-    def supports(self, did: Union[str, DID]) -> bool:
+    async def supports(self, profile: Profile, did: Union[str, DID]) -> bool:
```

Comment:
With these changes, `supports` becomes a method that should be overridden by implementing resolvers. As such, we should do a similar interface change as before where we do not require plugins to concern themselves with PyDID unless they want to. So `supports` should take a DID represented as a `str` instead of a `Union[str, DID]`.

Reply:
understood. I will remove the strong typing. 
please comment on pr #84